src/libman/Commands/InstallCommand.cs

```diff
@@ -0,0 +1,223 @@
+                foreach (IError error in result.Errors)
+                {
+                    Logger.Log(string.Format("[{0}]: {1}", error.Code, error.Message), LogLevel.Error);
+                    if(error.Code == PredefinedErrors.LibraryCannotBeInstalledDueToConflicts("", new List<string>()).Code)
+                    {
+                        isFileConflicts = true;
+                    }
+                }
+
+                if (isFileConflicts)
+                {
+                    Logger.Log(Resources.SpecifyDifferentDestination, LogLevel.Error);
+                }
+            }
+            else
+            {
+                // Just output failed.
+            }
+
+            return 0;
+        }
+
+        private async Task<(string libraryId, ILibrary library)> ValidateLibraryExistsInCatalogAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                ILibrary libraryToInstall = await ProviderCatalog.GetLibraryAsync(LibraryId.Value, cancellationToken);
+
+                if (libraryToInstall != null)
+                {
+                    return (LibraryId.Value, libraryToInstall);
+                }
+            }
+            catch
+            {
+                // The library id wasn't in the exact format.
+            }
+
+            IReadOnlyList<ILibraryGroup> libraryGroup = await ProviderCatalog.SearchAsync(LibraryId.Value, 5, cancellationToken);
+
+            IError invalidLibraryError = PredefinedErrors.UnableToResolveSource(LibraryId.Value, ProviderId);
+            if (libraryGroup.Count == 0)
+            {
+                throw new InvalidOperationException($"[{invalidLibraryError.Code}]: {invalidLibraryError.Message}");
+            }
+
+            var sb = new StringBuilder();
+
+            foreach (ILibraryGroup libGroup in libraryGroup)
+            {
+                IEnumerable<string> libIds = await libGroup.GetLibraryIdsAsync(cancellationToken);
+                if (libIds == null || !libIds.Any())
+                {
+                    continue;
+                }
+
+                if (libGroup.DisplayName.Equals(LibraryId.Value, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Found a group with an exact match.
+                    string libraryId = libIds.First();
+                    ILibrary libraryToInstall = await ProviderCatalog.GetLibraryAsync(libraryId, cancellationToken);
+
+                    return (libraryId, libraryToInstall);
+                }
+
+                sb.AppendLine("  " + libIds.First());
+            }
+
+            sb.Insert(0, $"[{invalidLibraryError.Code}]: {invalidLibraryError.Message} {Environment.NewLine} {Resources.SuggestedIdsMessage}{Environment.NewLine}");
+            throw new InvalidOperationException(sb.ToString());
+        }
+
+        private void ValidateParameters(Manifest manifest)
+        {
+            var errors = new List<string>();
```

Comment:
We have predefined errors for most of the errors below. See extension method: IsValid(this ILibraryInstallationState state, out IEnumerable<IError> errors)